From a51a2d26707c19999624c56edbd7fb63890636b5 Mon Sep 17 00:00:00 2001
From: Zhenyu Liang <zhenyu.liang@intel.com>
Date: Wed, 7 May 2014 16:08:50 +0800
Subject: [PATCH] Fix a crash when creating an incognito tab on tablet

The KitKat's WebView doesn't support the incognito mode.
Opening an incognito tab leads to a crash.
So hide the menu item on table as while as PhoneUI.

Change-Id: I37b2842fc2fe2fc927226fc90b9ccf5b0328ef1b
---
 src/com/android/browser/XLargeUi.java |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/com/android/browser/XLargeUi.java b/src/com/android/browser/XLargeUi.java
index 0d9b00e..e2db5a1 100644
--- a/src/com/android/browser/XLargeUi.java
+++ b/src/com/android/browser/XLargeUi.java
@@ -24,6 +24,7 @@ import android.graphics.drawable.BitmapDrawable;
 import android.graphics.drawable.Drawable;
 import android.graphics.drawable.LayerDrawable;
 import android.graphics.drawable.PaintDrawable;
+import android.os.Build;
 import android.os.Bundle;
 import android.os.Handler;
 import android.util.Log;
@@ -124,6 +125,10 @@ public class XLargeUi extends BaseUi {
         if (bm != null) {
             bm.setVisible(false);
         }
+        MenuItem incognito = menu.findItem(R.id.incognito_menu_id);
+        if (incognito != null) {
+            incognito.setVisible(Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT);
+        }
         return true;
     }
 
-- 
1.7.10.4

